### PR TITLE
Feature/read requests

### PR DESCRIPTION
**Pull request by AnonJon**
Add Chainlink Read Requests
- requests on feeds daily
- requesters daily to see what protocol are using the feeds

### CODE REVIEW

**Comment by jeff-dude** [comment]
@Hosuke another one in the chainlink directory, should be similar to the last one 👍  thank you!

File: models/chainlink/arbitrum/chainlink_arbitrum_read_requests_feeds_daily.sql [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  'arbitrum' as blockchain,
+  MAX(CAST(date_trunc('month', date_start) as date)) as date_month,
+  CAST(date_trunc('day', date_start) AS date) AS date_start,
+  log_meta.feed_address,
+  MAX(feed_name) as feed_name,
+  COALESCE(COUNT(log_meta.feed_address), 0) as total_requests
+FROM {{ ref('chainlink_arbitrum_read_requests_feeds') }} log_meta
+{% if is_incremental() %}
+  WHERE
+   date_start >= date_trunc('day', now() - interval '{{incremental_interval}}' day)
```
Comment (Hosuke) [inline models/chainlink/arbitrum/chainlink_arbitrum_read_requests_feeds_daily.sql:25]: We can use `incremental_predicate` to rewrite all the incremental filters:
i.e.
```suggestion
   {{ incremental_predicate('date_start') }}
```

File: models/chainlink/arbitrum/chainlink_arbitrum_read_requests_requester_daily.sql [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  'arbitrum' as blockchain,
+  cast(date_trunc('day', read_requests_requester.date_start) AS date) AS date_start,
+  MAX(cast(date_trunc('month', read_requests_requester.date_start) AS date)) AS date_month,
+  read_requests_requester.requester_address as requester_address,
+  MAX(read_requests_requester.requester_name) as requester_name,
+  COUNT(*) as total_requests
+FROM
+  {{ref('chainlink_arbitrum_read_requests_requester')}} read_requests_requester
+{% if is_incremental() %}
+  WHERE read_requests_requester.date_start >= date_trunc('day', now() - interval '{{incremental_interval}}' day)
```
Comment (Hosuke) [inline models/chainlink/arbitrum/chainlink_arbitrum_read_requests_requester_daily.sql:25]: Same as above.

File: models/chainlink/avalanche_c/chainlink_avalanche_c_read_requests_feeds_daily.sql [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  'avalanche_c' as blockchain,
+  MAX(CAST(date_trunc('month', date_start) as date)) as date_month,
+  CAST(date_trunc('day', date_start) AS date) AS date_start,
+  log_meta.feed_address,
+  MAX(feed_name) as feed_name,
+  COALESCE(COUNT(log_meta.feed_address), 0) as total_requests
+FROM {{ ref('chainlink_avalanche_c_read_requests_feeds') }} log_meta
+{% if is_incremental() %}
+  WHERE
+   date_start >= date_trunc('day', now() - interval '{{incremental_interval}}' day)
```
Comment (Hosuke) [inline models/chainlink/avalanche_c/chainlink_avalanche_c_read_requests_feeds_daily.sql:25]: Same as above.

File: models/chainlink/avalanche_c/chainlink_avalanche_c_read_requests_requester_daily.sql [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  'avalanche_c' as blockchain,
+  cast(date_trunc('day', read_requests_requester.date_start) AS date) AS date_start,
+  MAX(cast(date_trunc('month', read_requests_requester.date_start) AS date)) AS date_month,
+  read_requests_requester.requester_address as requester_address,
+  MAX(read_requests_requester.requester_name) as requester_name,
+  COUNT(*) as total_requests
+FROM
+  {{ref('chainlink_avalanche_c_read_requests_requester')}} read_requests_requester
+{% if is_incremental() %}
+  WHERE read_requests_requester.date_start >= date_trunc('day', now() - interval '{{incremental_interval}}' day)
```
Comment (Hosuke) [inline models/chainlink/avalanche_c/chainlink_avalanche_c_read_requests_requester_daily.sql:25]: Same as above.

File: models/chainlink/bnb/chainlink_bnb_read_requests_feeds_daily.sql [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  'bnb' as blockchain,
+  MAX(CAST(date_trunc('month', date_start) as date)) as date_month,
+  CAST(date_trunc('day', date_start) AS date) AS date_start,
+  log_meta.feed_address,
+  MAX(feed_name) as feed_name,
+  COALESCE(COUNT(log_meta.feed_address), 0) as total_requests
+FROM {{ ref('chainlink_bnb_read_requests_feeds') }} log_meta
+{% if is_incremental() %}
+  WHERE
+   date_start >= date_trunc('day', now() - interval '{{incremental_interval}}' day)
```
Comment (Hosuke) [inline models/chainlink/bnb/chainlink_bnb_read_requests_feeds_daily.sql:25]: Same as above.

File: models/chainlink/bnb/chainlink_bnb_read_requests_requester_daily.sql [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  'bnb' as blockchain,
+  cast(date_trunc('day', read_requests_requester.date_start) AS date) AS date_start,
+  MAX(cast(date_trunc('month', read_requests_requester.date_start) AS date)) AS date_month,
+  read_requests_requester.requester_address as requester_address,
+  MAX(read_requests_requester.requester_name) as requester_name,
+  COUNT(*) as total_requests
+FROM
+  {{ref('chainlink_bnb_read_requests_requester')}} read_requests_requester
+{% if is_incremental() %}
+  WHERE read_requests_requester.date_start >= date_trunc('day', now() - interval '{{incremental_interval}}' day)
```
Comment (Hosuke) [inline models/chainlink/bnb/chainlink_bnb_read_requests_requester_daily.sql:25]: Same as above.

File: models/chainlink/ethereum/chainlink_ethereum_read_requests_feeds_daily.sql [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  'ethereum' as blockchain,
+  MAX(CAST(date_trunc('month', date_start) as date)) as date_month,
+  CAST(date_trunc('day', date_start) AS date) AS date_start,
+  log_meta.feed_address,
+  MAX(feed_name) as feed_name,
+  COALESCE(COUNT(log_meta.feed_address), 0) as total_requests
+FROM {{ ref('chainlink_ethereum_read_requests_feeds') }} log_meta
+{% if is_incremental() %}
+  WHERE
+   date_start >= date_trunc('day', now() - interval '{{incremental_interval}}' day)
```
Comment (Hosuke) [inline models/chainlink/ethereum/chainlink_ethereum_read_requests_feeds_daily.sql:25]: Same as above.

File: models/chainlink/ethereum/chainlink_ethereum_read_requests_requester_daily.sql [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  'ethereum' as blockchain,
+  cast(date_trunc('day', read_requests_requester.date_start) AS date) AS date_start,
+  MAX(cast(date_trunc('month', read_requests_requester.date_start) AS date)) AS date_month,
+  read_requests_requester.requester_address as requester_address,
+  MAX(read_requests_requester.requester_name) as requester_name,
+  COUNT(*) as total_requests
+FROM
+  {{ref('chainlink_ethereum_read_requests_requester')}} read_requests_requester
+{% if is_incremental() %}
+  WHERE read_requests_requester.date_start >= date_trunc('day', now() - interval '{{incremental_interval}}' day)
```
Comment (Hosuke) [inline models/chainlink/ethereum/chainlink_ethereum_read_requests_requester_daily.sql:25]: Same as above.

File: models/chainlink/optimism/chainlink_optimism_read_requests_feeds_daily.sql [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  'optimism' as blockchain,
+  MAX(CAST(date_trunc('month', date_start) as date)) as date_month,
+  CAST(date_trunc('day', date_start) AS date) AS date_start,
+  log_meta.feed_address,
+  MAX(feed_name) as feed_name,
+  COALESCE(COUNT(log_meta.feed_address), 0) as total_requests
+FROM {{ ref('chainlink_optimism_read_requests_feeds') }} log_meta
+{% if is_incremental() %}
+  WHERE
+   date_start >= date_trunc('day', now() - interval '{{incremental_interval}}' day)
```
Comment (Hosuke) [inline models/chainlink/optimism/chainlink_optimism_read_requests_feeds_daily.sql:25]: Same as above.

File: models/chainlink/optimism/chainlink_optimism_read_requests_requester_daily.sql [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  'optimism' as blockchain,
+  cast(date_trunc('day', read_requests_requester.date_start) AS date) AS date_start,
+  MAX(cast(date_trunc('month', read_requests_requester.date_start) AS date)) AS date_month,
+  read_requests_requester.requester_address as requester_address,
+  MAX(read_requests_requester.requester_name) as requester_name,
+  COUNT(*) as total_requests
+FROM
+  {{ref('chainlink_optimism_read_requests_requester')}} read_requests_requester
+{% if is_incremental() %}
+  WHERE read_requests_requester.date_start >= date_trunc('day', now() - interval '{{incremental_interval}}' day)
```
Comment (Hosuke) [inline models/chainlink/optimism/chainlink_optimism_read_requests_requester_daily.sql:25]: Same as above.

File: models/chainlink/polygon/chainlink_polygon_read_requests_feeds_daily.sql [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+  'polygon' as blockchain,
+  MAX(CAST(date_trunc('month', date_start) as date)) as date_month,
+  CAST(date_trunc('day', date_start) AS date) AS date_start,
+  log_meta.feed_address,
+  MAX(feed_name) as feed_name,
+  COALESCE(COUNT(log_meta.feed_address), 0) as total_requests
+FROM {{ ref('chainlink_polygon_read_requests_feeds') }} log_meta
+{% if is_incremental() %}
+  WHERE
+   date_start >= date_trunc('day', now() - interval '{{incremental_interval}}' day)
```
Comment (Hosuke) [inline models/chainlink/polygon/chainlink_polygon_read_requests_feeds_daily.sql:25]: Same as above.

File: models/chainlink/polygon/chainlink_polygon_read_requests_requester_daily.sql [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+  'polygon' as blockchain,
+  cast(date_trunc('day', read_requests_requester.date_start) AS date) AS date_start,
+  MAX(cast(date_trunc('month', read_requests_requester.date_start) AS date)) AS date_month,
+  read_requests_requester.requester_address as requester_address,
+  MAX(read_requests_requester.requester_name) as requester_name,
+  COUNT(*) as total_requests
+FROM
+  {{ref('chainlink_polygon_read_requests_requester')}} read_requests_requester
+{% if is_incremental() %}
+  WHERE read_requests_requester.date_start >= date_trunc('day', now() - interval '{{incremental_interval}}' day)
```
Comment (Hosuke) [inline models/chainlink/polygon/chainlink_polygon_read_requests_requester_daily.sql:25]: Same as above.

File: models/chainlink/polygon/chainlink_polygon_read_requests_requester_daily.sql [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+    alias=alias('read_requests_requester_daily'),
+    partition_by=['date_month'],
+    materialized='incremental',
+    file_format='delta',
+    incremental_strategy='merge',
+    unique_key=['date_start', 'requester_address']
+  )
+}}
+
+{% set incremental_interval = '7' %}
```
Comment (Hosuke) [inline models/chainlink/polygon/chainlink_polygon_read_requests_requester_daily.sql:13]: After `incremantal_predicate` used, we can remove these jinja variables.
```suggestion
```

**Review by Hosuke — CHANGES_REQUESTED** [review]

**Comment by AnonJon** [comment]
Requested changes have been added.

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @AnonJon

**Review by jeff-dude — APPROVED** [review]
thanks both!

**Pull request merged**